src/subscriptions.c

```diff
@@ -952,6 +954,24 @@ subscription_done(void)
  * *************************************************************************/
 
 /**
+ * Update incoming byte count
+ */
+void subscription_add_bytes_in(th_subscription_t *s, size_t in)
+{
+  atomic_add(&s->ths_bytes_in, in);
+  atomic_add_u64(&s->ths_total_bytes_in, in);
```

Comment:
I'm not really sure what you mean here. Aren't atomic operations synchronized by default? Plus in the snippet you wrote the `in` parameter isn't used at all.